core/object.go

```diff
@@ -365,6 +366,10 @@ func (obj *ModuleObject) installConstructor(ctx context.Context, dag *dagql.Serv
 				Server:       dag,
 			})
 		},
+		// cache constructor calls per client; a given client will hit cache when making the same call repeatedly
+		func(ctx context.Context, _ dagql.Object, _ map[string]dagql.Input, origDgst digest.Digest) (digest.Digest, error) {
+			return mixDgstWithClientID(ctx, origDgst)
+		},
```

Comment:
nit: it would be really nice to be able to use `CachePerClient` here as well - I guess it's the same as the difference between `NodeFunc` and `Func`.

Even if we can't abstract them nicely (I tried lol, for the middlewares in #9395, it's not easy), it might be nice to have this as a predefined function.

Reply:
Yeah I couldn't find a way to convince the go compiler with generics to have a single one, happy to add a separate pre-defined func for this since